scipy/stats/tests/test_stats.py

```diff
@@ -1068,6 +1068,9 @@ class TestCorrSpearmanr:
         your program has them.
     """
 
+    def setup_method(self):
```

Comment:
Is this threadsafe? Should the rng be created in each method?

Reply:
The tests in this file seem to arbitrarily use this rng, or create it in the test.

Reply:
https://github.com/scipy/scipy/pull/23739/files#r2409809480 I don't know. I asked that in my comments on the PR. Please ping someone who would know. 

There is supposed to be a new instance of the class created for each `test_` method, and this code runs to initialize each class. I don't see how there would be sharing between threads, but maybe.